test/unit/ngsiv2/provisioning/device-provisioning-api_test.js

```diff
@@ -0,0 +1,791 @@
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+'use strict';
+
+var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
+    utils = require('../../../tools/utils'),
+    should = require('should'),
+    nock = require('nock'),
+    request = require('request'),
+    moment = require('moment'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        logLevel: 'FATAL',
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041,
+            baseRoot: '/'
+        },
+        types: {},
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    };
+
+describe('Device provisioning API: Provision devices', function() {
+    beforeEach(function(done) {
+        nock.cleanAll();
+
+        iotAgentLib.activate(iotAgentConfig, function() {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/registrations', function(body) {
+                    var expectedBody = utils.readExampleFile('./test/unit/ngsiv2/examples' +
+                        '/contextAvailabilityRequests/registerProvisionedDevice.json');
+
+                    // Note that expired field is not included in the json used by this mock as it is a dynamic
+                    // field. The following code performs such calculation and adds the field to the subscription
+                    // payload of the mock.
+                    if (!body.expires)
+                    {
+                        return false;
+                    }
+                    else if (moment(body.expires, 'YYYY-MM-DDTHH:mm:ss.SSSZ').isValid())
+                    {
+                        expectedBody.expires = moment().add(moment.duration(iotAgentConfig.deviceRegistrationDuration));
+                        var expiresDiff = moment(expectedBody.expires).diff(body.expires, 'milliseconds');
+                        if (expiresDiff < 500) {
+                            delete expectedBody.expires;
+                            delete body.expires;
+
+                            return JSON.stringify(body) === JSON.stringify(expectedBody);
+                        }
+
+                        return false;
+                    }
+                    else {
+                        return false;
+                    }
+                })
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5